sklearn/covariance/outlier_detection.py

```diff
@@ -90,17 +90,23 @@ def predict(self, X):
             The values of the less outlying point's decision function.
 
         """
-        if self.threshold is None:
-            raise Exception("Please fit data before predicting")
+        check_is_fitted(self, 'threshold_')
         is_inlier = -np.ones(X.shape[0], dtype=int)
         if self.contamination is not None:
             values = self.decision_function(X, raw_values=True)
-            is_inlier[values <= self.threshold] = 1
+            is_inlier[values <= self.threshold_] = 1
         else:
             raise NotImplementedError("You must provide a contamination rate.")
 
         return is_inlier
 
+    @property
+    def threshold(self):
+        warnings.warn(("The threshold attribute is renamed to threshold_ from "
+                       "0.16 onwards and will be removed in 0.18"),
+                      DeprecationWarning, stacklevel=1)
+        return self.threshold_
```

Comment:
to be truly backward compatible, this should probably be ``return getattr(self, "threshold_", None)``